pretyping/unification.ml

```diff
@@ -1117,12 +1117,13 @@ let rec unify_0_with_initial_metas (sigma,ms,es as subst : subst0) conv_at_top e
       in
       try
       let opt' = {opt with with_types = false} in
-      let substn = Reductionops.Stack.fold2
-                           (fun s u1 u -> unirec_rec curenvnb pb opt' s u1 (substl ks u))
-                           (evd,ms,es) us2 us in
-      let substn = Reductionops.Stack.fold2
-                           (fun s u1 u -> unirec_rec curenvnb pb opt' s u1 (substl ks u))
-                           substn params1 params in
+      let fold u1 u s = unirec_rec curenvnb pb opt' s u1 (substl ks u) in
+      let foldl acc l1 l2 =
+        try List.fold_right2 fold l1 l2 acc
+        with Invalid_argument _ -> assert false (* check_conv_record ensures lengths coincide *)
```

Comment:
What's the point of a dead code catch here?

Reply:
Same problem as with exception-based APIs, i.e. there is no way to ensure that it won't change and that we will get a stray exception because `fold_right2` raises catchable exceptions (it should assert-fail).

Reply:
I'm not very convinced but let's move on